src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new BulldozerProducer().get();
+        } else if (type.equals(Truck.class)) {
+            return new TruckProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return new ExcavatorProducer().get();
+        }
+        return List.of();
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. It will make the code cleaner and reduce duplication. For instance, you could create a variable for the `MachineProducer` and then return the result of its `get` method.